lagom-cloud-object-storage-example/account-impl/src/main/java/com/lightbend/lagom/account/impl/readside/AccountExtractProcessor.java

```diff
@@ -17,6 +17,13 @@
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
 
+/**
+ * A read-side processor that subscribes to AccountEntity events and forward them to AccountExtractRepository.
+ * 
+ * This processor does not saves offsets and consumes the full event stream from scratch each time the application is restarted.
+ * This is done on purpose. The repository does not save to any database, but keeps accumulating Account Extracts in-memory before
+ * archiving in Cloud Object Storage. 
```

Comment:
Upon restart, will it generate duplicate extracts on COS? Or will existing extracts be rewriten?

Reply:
Extracts will be rewritten. The extract id and key in COS is deterministic, `${account_number}#${extract_number}`.

So, when restarting we will consume all events again and on each 5th transaction, the extract will be uploaded. It's supposed to be exactly the same as the existing one.